contracts/modules/BaseMinter.sol

```diff
@@ -267,15 +279,19 @@ abstract contract BaseMinter is IMinterModule {
      * @param to        The address to mint to.
      * @param quantity  The quantity of tokens to mint.
      * @param affiliate The affiliate (referral) address.
+     * @param tip       The amount of tip.
      */
     function _mint(
         address edition,
         uint128 mintId,
         address to,
         uint32 quantity,
-        address affiliate
+        address affiliate,
+        uint256 tip
```

Comment:
Why pass in a value instead of calculating it dynamically? 
`tip = msg.value - requiredEtherValue`

Reply:
If we are to implement dynamic pricing in the future, the price during submission and during mining can be different.

For that scenario, users will have to submit a slightly higher price than the spot price if they want their transaction to have a reasonably high rate of going through. The UI will suggest a suitable “slippage” for the user. And when the transaction is mined, the excess will be refunded to the user.